### PR TITLE
Add elm-review

### PR DESCRIPTION
**Pull request by MartinSStewart**
This completes https://github.com/guida-lang/compiler/issues/4. I disabled most of the default rules for now. I think it's a better idea to incrementally enable rules rather than try to manage one enormous diff in this PR. Also removing unused code across the entire repo probably isn't a good idea at this stage since there seems to be a lot of code that belongs to disabled parts of the compiler and presumably will be needed when re-enabling those parts later.

### CODE REVIEW

File: terminal/src/Develop/Socket.elm [hunk 1]
```diff
@@ -3,39 +3,40 @@ module Develop.Socket exposing (watchFile)
...
-      do  _ <- WS.receiveDataMessage connection
-          receiver
-
-    shutdown : SomeException -> IO ()
-    shutdown _ =
-      return ()
-  in
-    do  _pid <- forkIO (receiver `catch` shutdown)
-        pinger 1 `catch` shutdown
+a = 0
```
Comment (decioferreira) [inline terminal/src/Develop/Socket.elm:11]: do we need this?

File: terminal/src/Develop/Socket.elm [hunk 2]
```diff
@@ -3,39 +3,40 @@ module Develop.Socket exposing (watchFile)
...
+
+--watchFile : FilePath -> WS.PendingConnection -> IO ()
+--watchFile watchedFile pendingConnection =
```
Comment (decioferreira) [inline terminal/src/Develop/Socket.elm:14]: find it strange that we can comment out a function that is exposed, is this okay?

File: compiler/src/Reporting/Doc.elm [hunk 3]
```diff
@@ -99,7 +101,7 @@ encode : Doc -> E.Value
 encode doc =
     let
         _ =
-            Debug.log "doc" (Debug.toString (P.pretty 0 doc))
+            Debug.toString (P.pretty 0 doc)
```
Comment (decioferreira) [inline compiler/src/Reporting/Doc.elm:104]: not sure if we should remove this `Debug.log`, the `encode` function has not yet been implemented, and I'm not sure if we could convert this into a `Debug.todo`. Maybe we should consider making this an exception on the configuration?

**Review by decioferreira — CHANGES_REQUESTED** [review]
Thank you so much for this work @MartinSStewart 🙏❤️ I fully agree with this approach of disabled some of the rules for now and incrementally re-enabling them (thanks for thinking of this aspect).

I've made a few comments on the code, see if they all make sense.

I've also noticed that the format seems to be a little bit off, can you please confirm from your side by running something like `npx elm-format . --yes`?

Thanks again for your help!

File: terminal/src/Develop/Socket.elm (hunk 1 above)
Comment (MartinSStewart) [reply terminal/src/Develop/Socket.elm:11]: If we don't have one declaration in the module then elm-review won't be able to parse it. That's why I commented out the existing code.

File: terminal/src/Develop/Socket.elm (hunk 2 above)
Comment (MartinSStewart) [reply terminal/src/Develop/Socket.elm:14]: This module and a few others have a bunch of errors that prevent them from compiling. But they aren't used by the compiler so we don't get any errors. I just did the minimum to make it so elm-review won't failing when parsing them.

**Review by decioferreira — APPROVED** [review]

**Comment by decioferreira** [comment]
Thank you 💯

**Pull request merged**